GiniVision/Classes/Core/GiniScreenAPICoordinator+Camera.swift

```diff
@@ -0,0 +1,292 @@
+            case .image:
+                if let imageDocuments = self.visionDocuments as? [GiniImageDocument],
+                    let lastDocument = imageDocuments.last {
+                    if self.giniConfiguration.multipageEnabled {
+                        if let imageDocuments = self.visionDocuments as? [GiniImageDocument],
+                            lastDocument.isImported {
+                            self.showMultipageReview(withImageDocuments: imageDocuments)
+                        }
+                    } else {
+                        self.reviewViewController = self.createReviewScreen(withDocument: lastDocument)
+                        self.screenAPINavigationController.pushViewController(self.reviewViewController!,
+                                                                              animated: true)
+                    }
+                }
+            case .qrcode, .pdf:
+                self.analysisViewController = self.createAnalysisScreen(withDocument: firstDocument)
+                self.screenAPINavigationController.pushViewController(self.analysisViewController!,
+                                                                      animated: true)
+            }
+        }
+    }
+    
+}
+
+// MARK: - DocumentPickerCoordinatorDelegate
+
+extension GiniScreenAPICoordinator: DocumentPickerCoordinatorDelegate {
+    
+    func documentPicker(_ coordinator: DocumentPickerCoordinator,
+                        didPick documents: [GiniVisionDocument]) {
+        
+        self.validate(documents) { result in
+            switch result {
+            case .success(let validatedDocuments):
+                coordinator.dismissCurrentPicker {
+                    self.visionDocuments.append(contentsOf: validatedDocuments.map { $0.document })
+                    validatedDocuments.forEach { validatedDocument in
+                        if validatedDocument.error == nil {
+                            self.didCaptureAndValidate(validatedDocument.document)
+                        }
+                    }
+                    self.showNextScreenAfterPicking(documents: validatedDocuments.map { $0.document })
+                }
+            case .failure(let error):
+                var positiveAction: (() -> Void)?
+                
+                if let error = error as? FilePickerError {
+                    switch error {
+                    case .maxFilesPickedCountExceeded, .mixedDocumentsUnsupported:
+                        let imageDocuments = self.visionDocuments.flatMap { $0 as? GiniImageDocument }
+                        
+                        if imageDocuments.isNotEmpty {
+                            positiveAction = {
+                                coordinator.dismissCurrentPicker {
+                                    let imageDocuments = self.visionDocuments.flatMap { $0 as? GiniImageDocument }
+                                    self.showMultipageReview(withImageDocuments: imageDocuments)
```

Comment:
Same thing here.

Reply:
The same here. This changes a little bit in another branch that will be merged after this one.